solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -108,3 +221,21 @@ impl TokenTransferValidationContext for IbcStorage<'_, '_> {
         Ok(())
     }
 }
+
+fn get_account_info_from_key<'a, 'b>(accounts: &'a Vec<AccountInfo<'b>>, key: Pubkey) -> Result<&'a AccountInfo<'b>, TokenTransferError> {
+    accounts
+    .iter()
+    .find(|account| account.key == &key)
+    .ok_or(TokenTransferError::ParseAccountFailure)
+}
+
+fn check_amount_overflow(amount: Amount) -> Result<(), TokenTransferError> {
```

Comment:
This should return `u64` on success so the caller can use this function to convert to `u64`.